test/unit/org/apache/cassandra/db/commitlog/CommitLogSegmentManagerCDCTest.java

```diff
@@ -382,7 +380,7 @@ public String toString()
         public boolean equals(Object other)
         {
             CDCIndexData cid = (CDCIndexData)other;
-            return fileName.equals(cid.fileName) && offset == cid.offset;
+            return fileName.equals(cid.fileName) && offset == cid.offset && this.getClass().equals(other.getClass());
```

Comment:
Do we need to add this class check? That check happens after a cast, so we know that both objects are instances of `CDCIndexData`. Since we never extend the class, the compared objects always have the same class. In fact, `CDCIndexData` can be a final class. I think the class check that the IDE warning is asking for should be done before casting. For example:
```java
@Override
public boolean equals(Object other)
{
    if (!(other instanceof CDCIndexData))
        return false;
    CDCIndexData cid = (CDCIndexData)other;
    return fileName.equals(cid.fileName) && offset == cid.offset;
}
```

Reply:
Right you are!